pkg/kubelet/cm/cpumanager/topology/topology.go

```diff
@@ -44,22 +44,22 @@ type CPUTopology struct {
 	CPUDetails   CPUDetails
 }
 
-// CPUsPerCore returns the number of logical CPUs are associated with
+// MaxCPUsPerCore returns the max number of logical CPUs are associated with
 // each core.
-func (topo *CPUTopology) CPUsPerCore() int {
+func (topo *CPUTopology) MaxCPUsPerCore() int {
```

Comment:
let's add to the docstring something like: "when SMT (HyperThreading) is enabled but some logical cores are disabled, we have uneven CPUs per Core."

I like the idea and I praise the effort here, but this is still not perfect, albeit certainly an improvement. This function now give us the detected cpus per core, but we would need rather the expected value. E.g. given a CPU model and SMT enable/disable status, we would authoritatively know that CPUs per Core must be 2 (or 3 or 4...) when SMT enabled and 1 with SMT disable.

Reply:
> let's add to the docstring something like: "when SMT (HyperThreading) is enabled but some logical cores are disabled, we have uneven CPUs per Core."
done

> I like the idea and I praise the effort here, but this is still not perfect, albeit certainly an improvement. This function now give us the detected cpus per core, but we would need rather the expected value. E.g. given a CPU model and SMT enable/disable status, we would authoritatively know that CPUs per Core must be 2 (or 3 or 4...) when SMT enabled and 1 with SMT disable.
yes, i think maybe `takeByTopologyNUMA*` needs redesign and refactor to care about real topology in the end. currently it works under certain preconditions. I provides a testcase in cpu_assignment_test.go showing it fails to give the best choice. offline CPU is just another failure scenario and this cr tries to patch.

Reply:
Thanks, this is fine for the purpose of this PR. We indeed need to address further improvements in future PR(s).